faucet/valve.py

```diff
@@ -164,6 +165,17 @@ def _dump_lag_status_event_sock(self):
                             'port_no': port.number,
                             'status': port.dyn_lacp_up}})
 
+    def _dump_learned_hosts_event_sock(self):
+        learned_macs = []
+        for vlan in self.dp.vlans.values():
+            for host in vlan.dyn_host_cache.values():
```

Comment:
This is losing information since which vlan it's associated with is being lost.  Solution would either to have a map by vlan, or have each entry have a set of vlans. So:

vlans[vlan_id].eth_srcs[sth_src] = ...

or

eth_srcs[eth_src]['vlans'] = [list_of_vlans]

Reply:
after looking at the L2_LEARN event, looks like vlanid is a field that can/just be included here...

Reply:
ok. Changing that.

Reply:
@grafnu PTAL.